File: app/src/main/java/com/sattar/j/smsfake/tools/customViews/CustomTextView.java
package com.sattar.j.smsfake.tools.customViews;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Typeface;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;

import android.util.AttributeSet;

import com.sattar.j.smsfake.R;
import com.sattar.j.smsfake.SmsFakeApplication;

public class CustomTextView extends AppCompatTextView {

    private TypedArray textViewCustomFont;
    private int styleFont;

    public CustomTextView(Context context) {
        super(context);
        initFont(null);
    }

    public CustomTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initFont(attrs);
    }

    public CustomTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initFont(attrs);
    }

    private void initFont(AttributeSet attrs) {
        textViewCustomFont = getContext().obtainStyledAttributes(attrs, R.styleable.CustomFont);
        styleFont = textViewCustomFont.getInt(R.styleable.CustomFont_styleFont, 0);
        Typeface typeface = Typeface.createFromAsset(getContext().getAssets()
                , SmsFakeApplication.getFont(styleFont));
        setTypeface(typeface);
    }

//    public void initFont(@Nullable String type) {
//
//    }
}

File: app/src/main/java/com/sattar/j/smsfake/view/customViews/CustomTextView.kt
package com.sattar.j.smsfake.view.customViews

import android.content.Context
import android.content.res.TypedArray
import android.graphics.Typeface
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.sattar.j.smsfake.R
import com.sattar.j.smsfake.SmsFakeApplication.Companion.getFont

class CustomTextView : AppCompatTextView {
    private var textViewCustomFont: TypedArray? = null
    private var styleFont = 0

    constructor(context: Context?) : super(context!!) {
        initFont(null)
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {
        initFont(attrs)
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {
        initFont(attrs)
    }

    private fun initFont(attrs: AttributeSet?) {
        textViewCustomFont = context.obtainStyledAttributes(attrs, R.styleable.CustomFont)
        styleFont = textViewCustomFont!!.getInt(R.styleable.CustomFont_styleFont, 0)
        val typeface = Typeface.createFromAsset(context.assets, getFont(styleFont))
        setTypeface(typeface)
    }
}